grblas/base.py

```diff
@@ -125,13 +233,128 @@ def new(self, *, dtype=None, mask=None):
         else:
             output = self.output_constructor()
         if mask is None:
-            mask = slice(None)  # [:] indicates no mask
-        elif not isinstance(mask, output.__class__):
-            raise TypeError(f'Mask must be type {output.__class__}')
-        output[mask] = self
+            output.update(self)
+        else:
+            if not isinstance(mask, output.__class__):
+                raise TypeError(f'Mask must be type {output.__class__}')
+            output(mask).update(self)
         return output
 
 
+class AmbiguousAssignOrExtract:
+    def __init__(self, parent, resolved_indexes):
+        self.parent = parent
+        self.resolved_indexes = resolved_indexes
+
+    def __call__(self, *args, **kwargs):
+        # Occurs when user calls C[index](params)
+        # Reverse the call order so we can parse the call args and kwargs
+        updater = self.parent(*args, **kwargs)
+        return updater[self.resolved_indexes]
+
+    def __lshift__(self, obj):
+        # Occurs when user calls C(params)[index] << obj or C[index] << obj
+        # Delegate back to parent's __setitem__ method
+        self.parent[self.resolved_indexes] = obj
+
+    def update(self, obj):
+        # Occurs when user calls C(params)[index].update(obj) or C[index].update(obj)
+        # Delegate back to parent's __setitem__ method
+        self.parent[self.resolved_indexes] = obj
+
+    @property
+    def value(self):
+        if isinstance(self.parent, Updater):
+            raise TypeError('Cannot extract from an Updater')
```

Comment:
Does this mean we can't do `my_matrix()[0, 0].value`?  What is this trying to guard against?